File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPTicksLived.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPTicksLived implements PlayerStateProperty<Integer> {
    @Override
    public String getConfigName() {
        return "ticks_lived";
    }

    @Override
    public Integer getDefaultSerializedProperty() {
        return 1;
    }

    @Override
    public Integer getSerializedPropertyFromPlayer(Player player) {
        return player.getTicksLived();
    }

    @Override
    public void restoreProperty(Player player, Integer ticks) {
        player.setTicksLived(ticks);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPTicksLived.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPTicksLived extends PlayerStateProperty[Int]:
  override def getConfigName: String = "ticks_lived"

  override def getDefaultSerializedProperty: Int = 1

  override def getSerializedPropertyFromPlayer(player: Player): Int =
    player.getTicksLived

  override def restoreProperty(player: Player, ticks: Int): Unit =
    player.setTicksLived(ticks)